src/components/Trade/TradeInputV2.tsx

```diff
@@ -28,54 +39,91 @@ import { type TradeState, TradeAmountInputField, TradeRoutePaths } from './types
 
 export const TradeInput = () => {
-  useSwapperService()
+  const { isLoadingTradeQuote, isLoadingFiatRateData } = useSwapperService()
   const [isLoading, setIsLoading] = useState(false)
-  const { setTradeAmounts } = useTradeAmounts()
-  const { checkApprovalNeeded, getTrade } = useSwapper()
+  const { setTradeAmountsAsynchronous, setTradeAmountsSynchronous } = useTradeAmounts()
+  const { checkApprovalNeeded, getTrade, bestTradeSwapper } = useSwapper()
   const history = useHistory()
   const borderColor = useColorModeValue('gray.100', 'gray.750')
   const { control, setValue, getValues, handleSubmit } = useFormContext<TradeState<KnownChainIds>>()
+  const {
+    state: { wallet },
+  } = useWallet()
 
+  // Watched form fields
   const sellTradeAsset = useWatch({ control, name: 'sellTradeAsset' })
   const buyTradeAsset = useWatch({ control, name: 'buyTradeAsset' })
   const quote = useWatch({ control, name: 'quote' })
   const feeAssetFiatRate = useWatch({ control, name: 'feeAssetFiatRate' })
+  const buyAssetFiatRate = useWatch({ control, name: 'buyAssetFiatRate' })
   const fees = useWatch({ control, name: 'fees' })
   const sellAssetAccountId = useWatch({ control, name: 'sellAssetAccountId' })
   const buyAssetAccountId = useWatch({ control, name: 'buyAssetAccountId' })
   const fiatSellAmount = useWatch({ control, name: 'fiatSellAmount' })
   const fiatBuyAmount = useWatch({ control, name: 'fiatBuyAmount' })
   const slippage = useWatch({ control, name: 'slippage' })
 
-  const translate = useTranslate()
-
+  // Selectors
   const sellFeeAsset = useAppSelector(state =>
     selectFeeAssetById(state, sellTradeAsset?.asset?.assetId ?? ethAssetId),
   )
-  const sellAssetBalance = useAppSelector(state =>
+  const feeAssetBalance = useAppSelector(state =>
+    sellFeeAsset
+      ? selectPortfolioCryptoHumanBalanceByAssetId(state, { assetId: sellFeeAsset?.assetId })
+      : null,
+  )
+  const sellAssetBalanceCrypto = useAppSelector(state =>
     selectPortfolioCryptoBalanceByFilter(state, {
       accountId: sellAssetAccountId,
       assetId: sellTradeAsset?.asset?.assetId ?? '',
     }),
   )
 
-  const protocolFee = fromBaseUnit(bnOrZero(fees?.tradeFee), buyTradeAsset?.asset?.precision ?? 0)
-  const toCryptoAmountAfterFees = bnOrZero(buyTradeAsset?.amount).minus(bnOrZero(protocolFee))
+  const sellAssetBalanceHuman = useAppSelector(state =>
+    selectPortfolioCryptoHumanBalanceByFilter(state, {
+      accountId: sellAssetAccountId,
+      assetId: sellTradeAsset?.asset?.assetId ?? '',
+    }),
+  )
+
+  // Constants
+  const sellAssetAccountSupported = (() => {
+    const sellAsset = sellTradeAsset?.asset
+    if (sellAsset && isSupportedNonUtxoSwappingChain(sellAsset.chainId)) return true
+    if (sellAsset && isSupportedUtxoSwappingChain(sellAsset.chainId) && sellAssetAccountId) {
+      const { utxoParams } = getUtxoParams(sellAssetAccountId)
+      return !!utxoParams?.bip44Params
+    }
+    return false
+  })()
```

Comment:
This looks off to me, this brings "general" notion of a supported sell asset, but no wallet-specific one. If the intent was only to check for support for trade-getting, it would work, but if the intent of this is displaying ``%{sellAssetSymbol} not supported by wallet` as specced in https://github.com/shapeshift/web/issues/2852, that's not going to have the intended behavior.

Suggestions:

- Rename to make it clear what `supported` means (in this context, that would be "by wallet, for trade" which be a bit too verbose)
- Bring wallet feature-detection over to this expression

Reply:
Addressed in https://github.com/shapeshift/web/pull/2789/commits/0211c4303224a3fcd18c5c8dab6601ef8154df6f, solid observation ser.